fabric-common-tags-api-v1/src/datagen/java/net/fabricmc/fabric/impl/tag/common/datagen/generators/BlockTagGenerator.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.tag.common.datagen.generators;
+
+import net.minecraft.block.Blocks;
+import net.minecraft.tag.BlockTags;
+
+import net.fabricmc.fabric.api.datagen.v1.FabricDataGenerator;
+import net.fabricmc.fabric.api.datagen.v1.provider.FabricTagProvider;
+import net.fabricmc.fabric.api.tag.v1.CommonBlockTags;
+
+public class BlockTagGenerator extends FabricTagProvider.BlockTagProvider {
+	public BlockTagGenerator(FabricDataGenerator dataGenerator) {
+		super(dataGenerator);
+	}
+
+	@Override
+	protected void generateTags() {
+		getOrCreateTagBuilder(CommonBlockTags.IRON_ORES)
+				.addOptionalTag(BlockTags.IRON_ORES);
+		getOrCreateTagBuilder(CommonBlockTags.NETHERITE_ORES)
+				.add(Blocks.ANCIENT_DEBRIS);
```

Comment:
Ancient debris cannot be considered a netherite ore.

Reply:
What would you consider it then? Its the block you mine and smelt to get netherite.

Reply:
At most, ancient debris can be considered netherite scrap ore; as with traditional ores, the ore can be smelted to get the raw material.

However, ancient debris differs in several key characteristics from ores:

- Obviously, ancient debris does not have 'ore' in its name
- As of Minecraft 1.17, no ores drop themselves by default (despite ancient debris still doing so)
- Ancient debris lacks an obvious substrate, unlike stone, deepslate, and netherrack ores
- Netherite ingots must be crafted rather than obtained through loot or smelting
- Ancient debris is not an instance of `OreBlock`
- Ancient debris does not drop experience

One argument in favor of ancient debris as ore is that the configured feature identifier for ancient debris is `minecraft:ore_ancient_debris_<small/large>`. However, several clearly non-ore blocks use this name scheme as well, namely including magma, gravel, dirt, and infested stone.

Reply:
Neither is RedstoneOre an instance of OreBlock, but alright.

Reply:
While redstone ore is not an instance of `OreBlock`, it does still meet the other qualifications listed above. In particular, the `RedstoneOreBlock` class reimplements experience drops in the same way as the `OreBlock` class does. Another thing to note is that the Deep Dark experimental snapshot removes `OreBlock` and the experience dropping in `RedstoneOreBlock` entirely.